[Docs] Add VSCode guide to docs (#5124)
qmk/qmk_firmware: docs/other_vscode.md

# Setting up Visual Studio Code for QMK Development

> Visual Studio Code ([VS Code](https://code.visualstudio.com/) is an open-source [Integrated Development Environment](https://en.wikipedia.org/wiki/Integrated_development_environment) (IDE), that supports many different programming languages. 

Reviewer: Although VS Code blurs the line between editor and IDE a bit, it's not an IDE, since it mainly accomplishes this through extensions. Microsoft doesn't call it an IDE either; they insist that it's a code editor.

```suggestion
[Visual Studio Code](https://code.visualstudio.com/) (VS Code) is an open-source code editor that supports many different programming languages. 
```

Also, I think the link should go on the full name.

> Using an IDE such as VS Code  provides many advantages over a plain text editor, such as:

Reviewer: ```suggestion
Using a full-featured editor such as VS Code provides many advantages over a plain text editor, such as:
```

* intelligent code completion
* convenient navigation in the code
* refactoring tools
* build automation (no need for the command-line)
* a graphical front end for GIT
* many other tools such as debugging, code formatting, showing call hierarchies etc.

> The purpose of this page is to document how to set-up VS Code for developing QMK Firmware.

Reviewer: ```suggestion
The purpose of this page is to document how to set up VS Code for developing QMK Firmware.
```

> The purpose of the is page is to document how to set-up VS Code for developing QMK Firmware.

Reviewer 2: ```suggestion
The purpose of this page is to document how to set-up VS Code for developing QMK Firmware.
```

  
  1. Disable all of the options but `Git LFS (Large File Support)` and `Check daily for Git for Windows updates`. 
  2. Set the default editor to `Use Visual Studio Code as Git's default editor`
  3. Select the `Use Git from Git Bash only` option, since that's the option that you should use here.
  4. For the `Choosing HTTPS transport backend`, either option should be fine.
  5. Select the `Checkout as-is, commit Unix-style line endings` option. QMK Firmware uses Unix style commits.
  6. For the extra options, leave the default options as is. 

  This software is needed for Git support in VS Code. It may be possible to not include this, but it is much simpler to just use this. 

* [Git Credential Manager for Windows](https://github.com/Microsoft/Git-Credential-Manager-for-Windows/releases) (Optional) 

  This software provides better support for Git by providing secure storage for git credentials, MFA and personal access token generation. 
  
  This isn't strictly needed, but we would recommend it. 

### Installing VS Code

1. Head to [VS Code](https://code.visualstudio.com/) and download the installer
2. Run the installer

> This part is super simple.  However, there is some configuration that we need to do to make things are configured correctly.

Reviewer 2: ```suggestion
This part is super simple.  However, there is some configuration that we need to do to ensure things are configured correctly.
```

### Configuring VS Code

First, we need to set up IntelliSense. This isn't strictly required, but it will make your life a LOT easier. To do this, we need to create the `.vscode/c_cpp_properies.json` file in the QMK Firmware folder, You can do this all manually, but I've done most of the work already. 

Grab [this file](https://gist.github.com/drashna/48e2c49ce877be592a1650f91f8473e8) and save it.  You may need to edit this file, if you didn't install MSYS2 to the default location, or are using WSL/LxSS.  

Once you have saved this file, you will need to reload VS Code, if it was already running. 

?> You should see an `extensions.json` and `settings.json` file in the `.vscode` folder, as well.

Now, we will set up the MSYS2 window to show up in VSCode as the integrated terminal.  This has a number of advantages. Mostly, you can control+click on errors and jump to those files.  This makes debugging much easier.  It's also nice, in that you don't have to jump to another window. 

> 1. Click <kbd><kbd>File</kbd> > <kbd>Preferences ></kbd> > <kbd>Preferences</kbd> </kbd>

Reviewer 3: That should probably read:
File > Preferences > Settings

Author: haha, yeah, you're right! 

(next time, you can hit "ctrl+g" to create a "suggestion" on git, and that can be directly applied here)

```suggestion
1. Click <kbd><kbd>File</kbd> > <kbd>Preferences ></kbd> > <kbd>Settings</kbd> </kbd>
```

> 2. Click on the <kbd>{}</kbd> button, in the top right to open the `settings.json` file. 

Reviewer 3: On windows, this is the ... button

Author: I'm on Windows, and this is `{}` on my system. 
![image](https://user-images.githubusercontent.com/2584963/52762423-5bb60480-2fcc-11e9-9823-5a14292f68aa.png)

The elipse button doesn't show this option.  I am on 1.31.1, on Windows 10.

Reviewer 3: Ah, version difference explains it, just updated and there it is

Author: Ah, glad to hear it. :)
And thanks for the feedback!
